gui/qt/external_plugins_window.py

```diff
@@ -0,0 +1,451 @@
+    ExternalPluginCodes.INSTALLED_BUT_FAILED_LOAD: _("The plugin is installed, but in the process of enabling and loading it, an error occurred. Restart Electron Cash and try again, or uninstall it and report it to it's developers."),
+    ExternalPluginCodes.INCOMPATIBLE_VERSION: _("The plugin is targeted at a later version of Electron Cash."),
+}
+
+use_tree_widget = True
+
+class FixedCheckBox(QCheckBox):
+    def mousePressEvent(self, event):
+        event.ignore()
+        
+    def mouseReleaseEvent(self, event):
+        event.ignore()
+
+
+class ExternalPluginsPreviewDialog(WindowModalDialog):
+    def __init__(self, plugin_dialog, main_window, title, plugin_path):
+        WindowModalDialog.__init__(self, main_window, title)
+
+        self.main_window = main_window
+        self.plugin_dialog = plugin_dialog
+
+        self.setMinimumWidth(400)
+        self.setMaximumWidth(400)
+
+        vbox = QVBoxLayout()
+        self.setLayout(vbox)
+
+        self.metadataFormLayout = QFormLayout()
+        self.pluginNameLabel = QLabel()
+        self.metadataFormLayout.addRow(_("Name"), self.pluginNameLabel)
+        self.versionLabel = QLabel()
+        self.metadataFormLayout.addRow(_("Version"), self.versionLabel)
+        self.descriptionLabel = QLabel()
+        self.descriptionLabel.setWordWrap(True)
+        self.metadataFormLayout.addRow(_("Description"), self.descriptionLabel)
+        self.supportedInterfacesLayout = QVBoxLayout()
+        self.supportedInterfacesLabel = QLabel(_("Integration"))
+        self.supportedInterfacesLabel.setAlignment(Qt.AlignLeft | Qt.AlignTop)
+        self.metadataFormLayout.addRow(self.supportedInterfacesLabel, self.supportedInterfacesLayout)
+        
+        self.qtInterfaceCheckBox = FixedCheckBox("User interface.")
+        self.supportedInterfacesLayout.addWidget(self.qtInterfaceCheckBox)
+        self.cmdLineInterfaceCheckBox = FixedCheckBox("Command-line.")
+        self.supportedInterfacesLayout.addWidget(self.cmdLineInterfaceCheckBox)
+
+        groupBox = QGroupBox(_("Plugin Metadata"))
+        groupBox.setLayout(self.metadataFormLayout)
+        vbox.addWidget(groupBox)
+
+        securityFormLayout = QFormLayout()
+        self.checksumLabel = QLabel()
+        self.checksumLabel.setAlignment(Qt.AlignRight)
+        self.checksumLabel.setToolTip(_("If the official source for this plugin has a checksum for this plugin, ensure that the value shown here is the same."))
+        securityFormLayout.addRow(_("Archive MD5 Checksum"), self.checksumLabel)
+
+        groupBox = QGroupBox(_("Security"))
+        groupBox.setLayout(securityFormLayout)
+        vbox.addWidget(groupBox)
+
+        confirmLayout = QVBoxLayout()
+        confirmLayout.setAlignment(Qt.AlignHCenter)
+        confirmGroupBox = QGroupBox()
+        self.liabilityCheckbox = QCheckBox(_("I understand and accept the risk involved in installing this plugin."))
```

Comment:
For me the textbox was too small (I couldn't read the full message, I had to expand the window)